packages/react-dom/src/client/ReactDOMHostConfig.js

```diff
@@ -670,44 +670,40 @@ export function registerSuspenseInstanceRetry(
   instance._reactRetry = callback;
 }
 
-export function getNextHydratableSibling(
-  instance: HydratableInstance,
-): null | HydratableInstance {
-  let node = instance.nextSibling;
+function getNextHydratable(node) {
   // Skip non-hydratable nodes.
-  while (
-    node &&
-    node.nodeType !== ELEMENT_NODE &&
-    node.nodeType !== TEXT_NODE &&
-    (!enableSuspenseServerRenderer ||
-      node.nodeType !== COMMENT_NODE ||
-      ((node: any).data !== SUSPENSE_START_DATA &&
-        (node: any).data !== SUSPENSE_PENDING_START_DATA &&
-        (node: any).data !== SUSPENSE_FALLBACK_START_DATA))
-  ) {
-    node = node.nextSibling;
+  for (; node != null; node = node.nextSibling) {
+    const nodeType = node.nodeType;
+    if (nodeType === ELEMENT_NODE || nodeType === TEXT_NODE) {
+      break;
+    }
+    if (enableSuspenseServerRenderer) {
+      if (nodeType === COMMENT_NODE) {
+        break;
```

Comment:
This should actually have been `continue`. Logic is hard. :)

Reply:
Also, test coverage is hard :D